<?xml version="1.0" encoding="utf-8"?>

<launch>

  <!-- Name of the robot description parameter -->
  <arg name="description_name" default="smb_description_for_model"/>
  <!-- TF prefix -->
  <arg name="tf_prefix" default=""/>
  <!-- Joint state topic name -->
  <arg name="joint_states_topic" default="/state_estimator/joint_states"/>
  <!-- Namespace to run the nodes in -->
  <arg name="ns" default="smb_tf_publisher"/>
  <arg name="publish_frequency" default="100" />
  <!-- Topic of the smb state -->
  <arg name="smb_state_subscriber_name" default="/state_estimator/smb_state"/>
  <!-- Publish map transforms -->
  <arg name="publish_map_transforms" default="false"/>

  <!-- Publish robot state (base) as TF -->
  <node ns="$(arg ns)" name="smb_tf_publisher" pkg="smb_tf_publisher" type="smb_tf_publisher_node" output="screen">
    <param name="subscribers/smb_state/topic"      value="/state_estimator/smb_state"/>
    <param name="base_frame_id"                     value="base"/>
    <param name="base_frame_id_prefix"              value=""/>
    <param name="ignore_state"                      value="false"/>
    <param name="subscribers/smb_state/topic"      value="$(arg smb_state_subscriber_name)"/>
    <param name="subscribers/smb_state/queue_size" value="10"/>
    <param name="subscribers/smb_state/latch"      value="false"/>
    <param name="publishers/pose/topic"             value="pose"/>
    <param name="publishers/pose/latch"             value="false"/>
    <param name="publishers/pose/queue_size"        value="10"/>
    <param name="publishers/joint_state/topic"      value="joint_state"/>
    <param name="publishers/joint_state/latch"      value="false"/>
    <param name="publishers/joint_state/queue_size" value="10"/>
    <param name="publish_frequency"                 value="$(arg publish_frequency)"/>
    <param name="tf_prefix"                         value="$(arg tf_prefix)"/>
    <param name="robot_description"                 value="$(arg description_name)"/>
    <param name="publish_map_transforms"                 value="$(arg publish_map_transforms)"/>
  </node>
  
</launch>
